### ISSUE TITLE
Cannot unlock a previously imported account

### ISSUE THREAD

**Issue by garethtdavies** [MinaProtocol/mina#9720]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/MinaProtocol/mina/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report. Please post those things in GitHub Discussions: https://github.com/MinaProtocol/mina/discussions

### Description

It is not possible to unlock a previously imported account. 

### Steps to Reproduce

1. Import the key

```mina accounts import --privkey-path /keys/my-wallet```

2. Confirm the account has been imported and is unlocked. I tested this importing with an environment variable and without. 

```mina accounts list```

3. Lock the account e.g. (or by restarting the node)

```mina accounts lock --public-key B62...```

4. Confirm it is now locked via:

```mina accounts list```

5. Try to unlock and it'll fail with `❌ Error: Wrong password provided (in ["unlockAccount"])`

Tested on multiple passwords (re-encryped a key) and while using env vars and manually entering the password.
...

### Expected Result

The password should unlock the account.

### Actual Result

`❌ Error: Wrong password provided (in ["unlockAccount"])`

### How frequently do you see this issue?

Always

### What is the impact of this issue on your ability to run a node?

High

### Status

```shell
root@d4ff686723f0:/data# mina client status
Mina daemon status
-----------------------------------

Global number of accounts:                     6228
Block height:                                  10569
Max observed block height:                     10569
Max observed unvalidated block height:         10569
Local uptime:                                  17h8m33s
Ledger Merkle root:                            jxdwZW3wh3ugkH7g372NDX4gMG5mQxRkjxF2LCntDfcdACSGSE8
Protocol state hash:                           3NKpxWqnM4JJL3ZQcifLf3bE5MLrzPBr3qXEyvLZXp3TfKbYmLb4
Chain id:                                      b6ee40d336f4cc3f33c1cc04dee7618eb8e556664c2b2d82ad4676b512a82418
Git SHA-1:                                     1f98b8be743e78993670e7b148be72c894b212e3
Configuration directory:                       /root/.mina-config
Peers:                                         46
User_commands sent:                            6
SNARK worker:                                  None
SNARK work fee:                                100000000
Sync status:                                   Synced
Catchup status:                                
        To build breadcrumb:           0
        To initial validate:           0
        Finished:                      362
        To verify:                     0
        To download:                   0
        Waiting for parent to finish:  0

Block producers running:                       0
Coinbase receiver:                             Block producer
Best tip consensus time:                       epoch=2, slot=2059
Best tip global slot (across all hard-forks):  16339
Consensus time now:                            epoch=2, slot=2060
Consensus mechanism:                           proof_of_stake
Consensus configuration:                       
        Delta:                     0
        k:                         290
        Slots per epoch:           7140
        Slot duration:             3m
        Epoch duration:            14d21h
        Chain start timestamp:     2021-09-24 00:00:00.000000Z
        Acceptable network delay:  3m
```

### Additional information

This is on ```mina version
Commit 1f98b8be743e78993670e7b148be72c894b212e3 on branch release/1.2.0```

**Comment by henry12322** [comment]
Contact coinbase support team https://my.forms.app/form/617e6cc70d06846f4ca75ee4

**Comment by areosps** [comment]
I'm seeing the exact same behavior. Unable to unlock the already imported account (tested on the interactive menu and using env_variable, same output).

It's quite a blocking bug if you cannot use your account.

Version 1.2.2:
Commit feee67cc2836bf12e792510a2440f28a0275740e on branch master

**Comment by garethtdavies** [comment]
For anyone else experiencing this (other than importing the wallet to another wallet such as Auro, which supports this keystore file type) you can also remove the wallet from the node, assuming you have the original by deleting the file from `~/.mina-config/wallets/store` and then restarting the daemon. After that you should be able to reimport the account and it will be unlocked.

**Comment by logan81k** [comment]
i got the exact same error.

> For anyone else experiencing this (other than importing the wallet to another wallet such as Auro, which supports this keystore file type) you can also remove the wallet from the node, assuming you have the original by deleting the file from `~/.mina-config/wallets/store` and then restarting the daemon. After that you should be able to reimport the account and it will be unlocked.

this is working for me. thanks
  
```
$ ~/.mina-config/wallets/store$ mina version
Commit feee67cc2836bf12e792510a2440f28a0275740e on branch master
```

**Comment by OV1-MINA** [comment]
It works for me too.

**Comment by andredunford** [comment]
Also having this issue. Running node in docker container.

`Error: Wrong password provided (in ["unlockAccount"])`

I've noticed that I am still able to perform other functions requiring authentication with `CODA_PRIVKEY_PASS` (e.g. `mina advanced dump-keypair `). The issue seems to be related directly to unlocking. 

Not sure it it is relevant, but I began to experience this issue sometime after updating to 1.2.2-feee67c-mainnet (docker), I had previous been running 1.1.5-a42bdee without issue.  However, now even if I revert to 1.1.5, I continue to get the error on unlock.

**Comment by innerfirexy** [comment]
I had the same issue on version "Commit feee67cc2836bf12e792510a2440f28a0275740e on branch master" on Ubuntu 18.04. @garethtdavies 's solution fixed it. Thanks.

**Comment by psteckler** [comment]
I can reproduce it locally.

**Comment by psteckler** [comment]
I've tracked down the immediate cause of the problem, still need to work out how we get into a bad state.

**Comment by psteckler** [comment]
Fixed by #10190.
